lib/logging/addon/mixins/logging-model.js

```diff
@@ -0,0 +1,99 @@
+import { get, set } from '@ember/object'
+import { inject as service } from '@ember/service'
+import { alias } from '@ember/object/computed'
+import Mixin from '@ember/object/mixin';
+
+const DEFAULT_TARGET_TYPE = 'none';
+export default Mixin.create({
+  // needs to override this type
+
+  scope: service(),
+  globalStore: service(),
+  pageScope: alias('scope.currentPageScope'),
+  cluster: alias('scope.currentCluster'),
+  project: alias('scope.currentProject'),
+  modalService: service('modal'),
+
+  kafka: {
```

Comment:
Why are all these set by default on every model?

Reply:
They exist because there are 5 logging targets and each of them needs to keep a logging configure of itself, or else they will override each other. And also createRecord does not create all the sub fields.